contracts/examples/dao/DAO.sol

```diff
@@ -49,83 +57,61 @@ contract DAO is VentureEth {
     }
 
     /**
-     * @notice Propose funding for venture. Can only be used after the original funding round. venture must be of type IssuanceEth.
-     * @param funding The amount to proposed as funding for the venture.
-     * @param venture The proposed venture. Must be of type IssuanceEth.
+     * @notice Propose a venture. Must have approved the DAO to spend gage of your DAO tokens. venture must inherit from VentureEth.
+     * @param venture The address of the VentureEth contract
+     * @param funding The ether to fund the venture with.
      */
-    function proposeVenture(uint256 funding, address venture) public {
-        require(currentState == "LIVE", "DAO needs to be LIVE.");
-        uint256 newFundingPool = fundingPool.add(funding);
-        require(
-            newFundingPool <= address(this).balance,
-            "Not enough funds."
-        );
-        fundingPool = newFundingPool;
-        proposedFundingForVenture[address(VentureEth(venture))] = funding;
-    }
-
-    /**
-     * @notice Vote for venture. Can only be used after the original funding round.
-     * @param votes The amount of tokens to lock for the venture
-     * @param venture The venture to vote for. Must be of type VentureEth.
-     */
-    function voteForVenture(uint256 votes, address venture) public {
-        require(currentState == "LIVE", "DAO needs to be LIVE.");
-        require(
-            this.balanceOf(msg.sender).sub(totalVotesByHolder[msg.sender]) >= votes,
-            "Not enough power."
-        );
-        totalVotesForVenture[venture] = totalVotesForVenture[venture].add(
-            votes
+    function proposeVenture(
+        address venture,
+        uint256 funding
+    ) public {
+        require(currentState == "LIVE", "DAO needs to be LIVE");
+        this.transferFrom(msg.sender, address(this), gage);
+        Voting voting = new Voting(address(this), threshold);
+        voting.registerProposal(
+            address(this),
+            abi.encodeWithSignature("fundVenture(address,uint256)", venture, funding)
         );
-        totalVotesByHolder[msg.sender] = totalVotesByHolder[msg.sender].add(
-            votes
+        voting.registerProposal(
```

Comment:
See below, we don't need to encode this in the proposal.

I definitely prefer to have as a pattern that `Voting.sol` takes a single contract call to be enacted. Using state machines is easy to avoid needed multiple `enact()` calls.